packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx

```diff
@@ -0,0 +1,208 @@
+import { TLSAuthSettings } from './TLSAuthSettings';
+import { DataSourceSettings } from '../../types';
+import { HttpSettingsProps } from './types';
+
+const ACCESS_OPTIONS: Array<SelectableValue<string>> = [
+  {
+    label: 'Server (default)',
+    value: 'proxy',
+  },
+  {
+    label: 'Browser',
+    value: 'direct',
+  },
+];
+
+const DEFAULT_ACCESS_OPTION = {
+  label: 'Server (default)',
+  value: 'proxy',
+};
+
+const HttpAccessHelp = () => (
+  <div className="grafana-info-box m-t-2">
+    <p>
+      Access mode controls how requests to the data source will be handled.
+      <strong>
+        <i>Server</i>
+      </strong>{' '}
+      should be the preferred way if nothing else stated.
+    </p>
+    <div className="alert-title">Server access mode (Default):</div>
+    <p>
+      All requests will be made from the browser to Grafana backend/server which in turn will forward the requests to
+      the data source and by that circumvent possible Cross-Origin Resource Sharing (CORS) requirements. The URL needs
+      to be accessible from the grafana backend/server if you select this access mode.
+    </p>
+    <div className="alert-title">Browser access mode:</div>
+    <p>
+      All requests will be made from the browser directly to the data source and may be subject to Cross-Origin Resource
+      Sharing (CORS) requirements. The URL needs to be accessible from the browser if you select this access mode.
+    </p>
+  </div>
+);
+
+export const DataSourceHttpSettings: React.FC<HttpSettingsProps> = props => {
+  const { defaultUrl, dataSourceConfig, onChange, showAccessOptions } = props;
+  let urlTooltip;
+  const [isAccessHelpVisible, setIsAccessHelpVisible] = useState(false);
+  const theme = useTheme();
+
+  const onSettingsChange = useCallback(
+    (change: Partial<DataSourceSettings<any, any>>) => {
+      onChange({
+        ...dataSourceConfig,
+        ...change,
+      });
+    },
+    [dataSourceConfig]
+  );
+
+  switch (dataSourceConfig.access) {
+    case 'direct':
+      urlTooltip = (
+        <>
+          Your access method is <em>Browser</em>, this means the URL needs to be accessible from the browser.
+        </>
+      );
+      break;
+    case 'proxy':
+      urlTooltip = (
+        <>
+          Your access method is <em>Server</em>, this means the URL needs to be accessible from the grafana
+          backend/server.
+        </>
+      );
+      break;
+    default:
+      urlTooltip = 'Specify a complete HTTP URL (for example http://your_server:8080)';
+  }
+
+  const accessSelect = (
+    <Select
+      width={20}
+      options={ACCESS_OPTIONS}
+      value={ACCESS_OPTIONS.filter(o => o.value === dataSourceConfig.access)[0] || DEFAULT_ACCESS_OPTION}
+      onChange={selectedValue => onSettingsChange({ access: selectedValue.value })}
+    />
+  );
+
+  const isValidUrl = /^(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?$/.test(
```

Comment:
@dprokop is there a reason not to use `validationEvents` prop on Inputs? Should that be in general used or is there some issues with that?

Reply:
@aocenas good question. I think we could use validationEvents in this case, can't see a reason why not. @peterholmberg do you wanna try it out?